in_toto/rulelib.go

```diff
@@ -0,0 +1,99 @@
+package in_toto
+
+import (
+  "fmt"
+  "strings"
+  )
+
+var errorMsg string = "Wrong rule format, available formats are:\n\t" +
+  "MATCH <pattern> [IN <source-path-prefix>] WITH (MATERIALS|PRODUCTS)" +
+      " [IN <destination-path-prefix>] FROM <step>,\n\t" +
+  "CREATE <pattern>,\n\t" +
+  "DELETE <pattern>,\n\t" +
+  "MODIFY <pattern>,\n\t" +
+  "ALLOW <pattern>,\n\t" +
+  "DISALLOW <pattern>\n\n"
+
+
+
+func UnpackRule(rule []string) (map[string]string, error) {
+  // Cache rule len
+  ruleLen := len(rule)
+
+  // Create all lower rule copy to case insensitively parse out tokens whose
+  // position we don't know yet
+  // We keep the original rule to retain the non-token elements' case
+  ruleLower := make([]string, ruleLen)
+  for i, val := range rule {
+    ruleLower[i] = strings.ToLower(val)
+  }
+
+  switch ruleLower[0] {
+    case "create", "modify", "delete", "allow", "disallow":
+        if ruleLen != 2 {
+          return nil,
+              fmt.Errorf("%s Got:\n\t %s", errorMsg, rule)
+        }
+
+        return map[string]string {
+          "type": ruleLower[0],
+          "pattern": rule[1],
+        }, nil
+
+    case "match":
+      var srcPrefix string
+      var dstType string
+      var dstPrefix string
+      var dstName string
+
+      if ruleLen == 10 && ruleLower[2] == "in" &&
```

Comment:
I'm not sure about hardcoded indices into a list. Consider using consts, like `IN_INDEX` and `WITH_INDEX`.

Reply:
Not sure if consts make sense, because the token's may have different positions in the list.
I hope these comments help: 486e53d027a54f2c94fd5eddc8553e2b8444cf72

Reply:
Yeah, I see, I too felt my suggestion wasn't very pretty.

A link to the spec perhaps makes sense, so that when this code gets updated, it's clear what version etc. this implementation supports.

Reply:
(My original comment was coming from the idea of future-proofing the field numbers.)

Reply:
It looks pretty much the same as in the reference implementation. While that is no argument for it, it is an indication that we haven't found (and needed) a better solution yet. Do you want to create a ticket, with an implementation suggestion? I'm not sure if I understand your suggestion here.